### ISSUE TITLE
Coffeescript bug: can't handle hex in range expression

### ISSUE THREAD

**Issue by greghuc** [DanielXMoore/Civet#342]
### Coffeescript bug (parse error)

```
"civet coffeeCompat"

x = [0...0xa]
```
gives
```
Unexpected non-whitespace character after JSON at position 1
```

### Playground links

[Coffeescript link](https://coffeescript.org/#try:x%20%3D%20%5B0...0xa%5D)
[Civet link](https://civet.dev/playground/?code=ImNpdmV0IGNvZmZlZUNvbXBhdCIKCnggPSBbMC4uLjB4YV0%3D)

**Comment by greghuc** [comment]
Forgot to say that using decimal in range works, and also that this bug happens in pure civet mode (without "civet coffeeCompat"):

```
"civet coffeeCompat"

x = [0...10]
```
gives
```
var x;
x = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
```

### Playground links

[Coffeescript link](https://coffeescript.org/#try:x%20%3D%20%5B0...10%5D)
[Civet link](https://civet.dev/playground/?code=ImNpdmV0IGNvZmZlZUNvbXBhdCIKCnggPSBbMC4uLjEwXQ%3D%3D)

**Comment by edemaine** [comment]
Thanks @greghuc for another helpful bug report! It's fixed on `main` now and will be released shortly.

**Comment by edemaine** [comment]
That's great news for our goal of CoffeeScript compatibility! Thanks for all the testing!

**Comment by greghuc** [comment]
I'm trying to get to a place where the coffeescript codebases could be migrated to civet. At present, I'm focussing on adding civet to projects, attempting to compile, and reporting bugs. This is a gradual addressing of technical debt, so I'm not working on it full time.

Then I need to think about civet in the context of the wider build system. Currently I'm using [coffeeify](https://github.com/jnordberg/coffeeify) as part of a wider browserify build, so that will need replacing somehow.

And finally, I'll need to do a gradual coffeescript->civet conversion. I'll likely start applying subtractions like "civet coffeeCompat -coffeeBooleans". I'm unclear on cli support for this at the moment - currently I'm running "civet -c -o" on a per file basis, and I notice that a .coffee filetype leads to "civet coffeeCompat" (without subtractions) being automatically applied.

**Comment by edemaine** [comment]
* We should probably write a `civetify` plugin in for browserify. (I already wrote one for legacy support for Gulp.) Let us know if you'd like to take that on. Looking at [coffeeify's index.js](https://github.com/jnordberg/coffeeify/blob/master/index.js), it shouldn't be too hard.
* `.coffee` files get `"civet coffeeCompat"` added automatically *only if there isn't already a `"civet"` directive alreadey*. So you can still use the subtractions. It will be easier once we have a global config file, or command-line options for turning on/off options...

**Comment by edemaine** [comment]
Update: This fix is now in v0.5.67 on NPM.
